system/Test/TestResponse.php

```diff
@@ -272,10 +272,19 @@ public function assertSessionHas(string $key, $value = null)
 	{
 		$this->assertTrue(array_key_exists($key, $_SESSION), "'{$key}' is not in the current \$_SESSION");
 
-		if ($value !== null)
+		if (is_null($value))
+		{
+			return;
+		}
+
+		if (is_string($value))
```

Comment:
Why not use `var_export` on`$value` to make it usable on the message?

Reply:
The first iteration had that but it just ends up duplicating the display and looks messy. I will paste my example result from that iteration.

Reply:
I found this made it difficult to determine what the actual message was:
```
There was 1 failure:

1) PresenterWriteTest::testCreate
The value of 'alerts-queue' (Array ( [0] => Array ( [class] => foobar [text] => New factory created successfully. ) ) ) does not match expected value.
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
     0 => Array (
-        'class' => 'foobar'
+        'class' => 'success'
         'text' => 'New factory created successfully.'
     )
 )
```

Reply:
Okay. Then how about using is_scalar to accommodate int, floats?

Reply:
That's a great idea! I'm mobile for the next 12 hours but if someone can suggest it I can commit the change (or you can).

Reply:
Need to confirm as well whether an object with `__toString()` will pass `is_scalar()`. If not we might want to check both scalar and string.

Reply:
Nevermind, `is_string` and `is_scalar` both always fail on objects - you have to check `is_object` and `method_exists('__toString')` to support that.